### PR TITLE
Added sanitize funtion to normalize the link input

### LINKED ISSUE #2546: We should normalize URL before saving it
# Current behavior
Screenshot 2023-11-16 at 16 52 08

# Expected behavior
Screenshot 2023-11-16 at 16 52 41

### PR DESCRIPTION
**Pull request by SujithThirumalaisamy**
**feat**: add sanitizeLinkRecordInput for link normalization

**Addresses issue #2546**

- Introduce a new utility function, **sanitizeLinkRecordInput**, to normalize link input specifically for the companies tab.
- This function is added to _packages/twenty-front/src/modules/object-record/utils/sanitizeLinkRecordInput.ts._
- Modify existing utility function **sanitizeRecordInput** in _packages/twenty-front/src/modules/object-record/utils/sanitizeRecordInput.ts_ to incorporate link sanitization logic.

**Fixes: #2546**

### CODE REVIEW

File: packages/twenty-front/src/modules/object-record/utils/sanitizeLinkRecordInput.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  recordObject: Record<string, any>,
+): Record<string, any> => {
+  return { ...recordObject, domainName: sanitizeLink(recordObject.domainName) };
+};
+const sanitizeLink = (link: string): string => {
+  let cleanedLink = link.replace(/^(ftp|http|https):\/\//, '');
+  cleanedLink = cleanedLink.replace(/\/+$/, '');
+  cleanedLink = cleanedLink.replace(/\/[^/]*$/, '');
+  return cleanedLink;
+};
```
Comment (thaisguigon) [inline packages/twenty-front/src/modules/object-record/utils/sanitizeLinkRecordInput.ts:11]: We also want to remove:
- `www.`
- search params after `?`

After testing `sanitizeLink`, here are some results I am obtaining:

image

Maybe we could implement something like this?

```suggestion
const getUrlHostName = (url: string) =>
  URL.canParse(url) ? new URL(url).hostname.replace(/^www\./, '') : '';

const sanitizeLink = (url: string) =>
  getUrlHostName(url) || getUrlHostName(`https://${url}`);
```

**Review by thaisguigon — CHANGES_REQUESTED** [review]
Hi @SujithThirumalaisamy!
Have you seen @charlesBochet's comment about how we want to address this issue? https://github.com/twentyhq/twenty/issues/2546#issuecomment-1845621874
This PR seems to address the third item, would be nice if we could address the other two in the same PR as well!
I can answer your questions if you need help!
Thanks a lot for your contribution 🙏 

**EDIT:** sorry I read @charlesBochet's check list too fast, he mentioned `persistLinkField` for the third item, not `sanitizeRecordInput`. @charlesBochet, is it ok if we normalize before saving as mentioned in the issue's title and as implemented in this PR? Or do we want to only normalize on field display?

**Comment by charlesBochet** [comment]
> Hi @SujithThirumalaisamy! Have you seen @charlesBochet's comment about how we want to address this issue? [#2546 (comment)](https://github.com/twentyhq/twenty/issues/2546#issuecomment-1845621874) This PR seems to address the third item, would be nice if we could address the other two in the same PR as well! I can answer your questions if you need help! Thanks a lot for your contribution 🙏
> 
> **EDIT:** sorry I read @charlesBochet's check list too fast, he mentioned `persistLinkField` for the third item, not `sanitizeRecordInput`. @charlesBochet, is it ok if we normalize before saving as mentioned in the issue's title and as implemented in this PR? Or do we want to only normalize on field display?

I actually think we should normalize before persisting the field (it should be normalize in DB)

**Comment by thaisguigon** [comment]
> > Hi @SujithThirumalaisamy! Have you seen @charlesBochet's comment about how we want to address this issue? [#2546 (comment)](https://github.com/twentyhq/twenty/issues/2546#issuecomment-1845621874) This PR seems to address the third item, would be nice if we could address the other two in the same PR as well! I can answer your questions if you need help! Thanks a lot for your contribution 🙏
> > **EDIT:** sorry I read @charlesBochet's check list too fast, he mentioned `persistLinkField` for the third item, not `sanitizeRecordInput`. @charlesBochet, is it ok if we normalize before saving as mentioned in the issue's title and as implemented in this PR? Or do we want to only normalize on field display?
> 
> I actually think we should normalize before persisting the field (it should be normalize in DB)

In this case this PR is on the right track! I would say there is no need to implement the check list items mentioned in https://github.com/twentyhq/twenty/issues/2546#issuecomment-1845621874 as normalizing `domainName` in `sanitizeRecordInput` should do the job.
@SujithThirumalaisamy the 2 comments I left above are still relevant, if you can address them it would be awesome 🙏

**Comment by SujithThirumalaisamy** [comment]
I have tested this jest tests in my local. Surprisingly I have got no fails. 
image
I am confused about the fail in the CI/CD. Is this normal @thaisguigon ?

**Review by thaisguigon — CHANGES_REQUESTED** [review]
Hi @SujithThirumalaisamy, thank you for your time!

I rebased your PR and fixed the conflicts!

One test was failing because of `URL.canParse`, it seems it doesn't work in Jest! As this is related to my suggestion, I pushed a commit to fix it, by using Zod instead of `URL.canParse`.

For the failing storybook tests (`IconPicker.stories.tsx`), this is unrelated to your PR, so don't worry about it!

I left a minor comment about a line that might be unnecessary, otherwise LGTM!

**Review by thaisguigon — APPROVED** [review]
Awesome, thank you!

**Pull request merged**